src/plugins/intel_gpu/src/graph/include/convolution_inst.h

```diff
@@ -132,6 +132,10 @@ struct typed_program_node<convolution> : public typed_program_node_base<convolut
             params->activations_zero_points_layout = optional_layout(activations_zero_points().get_output_layout());
         if (compensation_term())
             params->compensation_layout = optional_layout(compensation().get_output_layout());
+
+        params->groups = get_groups();
+        params->split = get_split();
+        params->deform_conv_dep_offset = get_deform_conv_dep_offset();
```

Comment:
Are you going to serialize primitive struct? If yes, these fields are not needed in kernel_impl_params.

Reply:
Anyways, kernel_impl_param doesn't look like a good place for primitive specific parameters, so need to think about better solution.

Reply:
> Are you going to serialize primitive struct?

No, I'm not going to serialize primitives. The goal of this PR is to remove dependencies to `program_node` when running execute_impl(). In the case of deserialized graph, `program` and `program_node` are null, and it makes errors when accessing `program_node`. So I copied the required values from `program_node` into `kernel_impl_params`.
In the original `convolution_inst.h`, the `groups`, `split`, and `deform_conv_dep_offset` were accessed through `program_node` as shown below. So I also copied them into the `kernel_impl_params'.

https://github.com/openvinotoolkit/openvino/blob/0aeb3d81517b7ce4285bff080d9c3e41d799f46f/src/plugins/intel_gpu/src/graph/include/convolution_inst.h#L160-L170

Now, I removed them from `kernel_impl_params` and copied them into `convolution_inst`.

Reply:
I think we'll need to serialize primitives for dynamic models later, so after that we can use it to get groups/split and other stuff.
As for current solution, I assume that you serialize typed primitive instances, so maybe those parameters can be a part of convolution_inst?

Reply:
Yes, you're right. I'm going to serialize typed primitive instances.